contracts/l2/fossil.cairo

```diff
@@ -15,91 +20,87 @@ end
 end
 
 func get_asset_data{
-    syscall_ptr : felt*, pedersen_ptr : HashBuiltin*, range_check_ptr
+    syscall_ptr : felt*, pedersen_ptr : HashBuiltin*, range_check_ptr, bitwise_ptr : BitwiseBuiltin*
 }(
-        block: felt,
-        account_160: felt,
-        slot: StorageSlot,
-        proof_sizes_bytes_len : felt,
-        proof_sizes_bytes : felt*,
-        proof_sizes_words_len : felt,
-        proof_sizes_words : felt*,
-        proofs_concat_len : felt,
-        proofs_concat : felt*
-) -> (
-    emissionPerSecond: felt,
-    index: felt,
-    lastUpdateTimestamp: felt
-):
+    block : Uint256,
+    proof_sizes_bytes_len : felt,
+    proof_sizes_bytes : felt*,
+    proof_sizes_words_len : felt,
+    proof_sizes_words : felt*,
+    proofs_concat_len : felt,
+    proofs_concat : felt*,
+) -> (index : Uint256):
+    alloc_locals
+
     # check the proof is for the right slot
-    let (slot_) = asset_data_slot.read()
-    with_attr error_message("Wrong storage slot"):
-        assert slot.word_1 = slot_.word_1
-        assert slot.word_2 = slot_.word_2
-        assert slot.word_3 = slot_.word_3
-        assert slot.word_4 = slot_.word_4
-    end
+    let (slot) = asset_data_slot.read()
+    let (account_160) = incentives_controller.read()
+    let (fossil_) = fossil.read()
 
     # check the proof is for a later block
     let (last_update_) = last_update.read()
-    let (le) = uint256_le(last_update_, last_update_)
+    let (le) = uint256_le(last_update_, block)
     with_attr error_message("Rejecting old block"):
         assert le = 1
     end
 
-    let (fossil_) = fossil.read()
-    let (asset_data: Uint256) = IFossil.get_storage_uint(
+    let (asset_data : Uint256) = IFossil.get_storage_uint(
         fossil_,
-        block,
+        block.low,
         account_160,
         slot,
         proof_sizes_bytes_len,
         proof_sizes_bytes,
         proof_sizes_words_len,
         proof_sizes_words,
         proofs_concat_len,
-        proofs_concat
+        proofs_concat,
     )
 
-    
+    # let mask_emission_per_second   = Uint256(low=0x00000000000000000000000000000000, high=0xffffffffffffffffffffffffff000000)
+    # let emission_per_second_rs     = Uint256(low=0x00000000000000000000000000000000, high=0x00000000000000000000000001000000)
+    let mask_index = Uint256(
+        low=0xffffffffffffffffffff000000000000, high=0x00000000000000000000000000ffffff
+    )
+    let index_rs = Uint256(
+        low=0x00000000000000000001000000000000, high=0x00000000000000000000000000000000
+    )
+    # let mask_last_update_timestamp = Uint256(low=0x00000000000000000000ffffffffff00, high=0x00000000000000000000000000000000)
+    # let last_update_timestamp_rs   = Uint256(low=0x00000000000000000000000000000100, high=0x00000000000000000000000000000000)
+
+    # let (masked_emission_per_second) = uint256_and(asset_data, mask_emission_per_second)
+    let (masked_index) = uint256_and(asset_data, mask_index)
```

Comment:
so far the `mask_index` & `masked_index` are correct, not sure about the `index_rs`